flink/src/main/java/org/apache/iceberg/flink/data/vectorized/ConstantColumnVectors.java

```diff
@@ -0,0 +1,217 @@
+    }
+  }
+
+  private static class ConstantLongColumnVongector implements LongColumnVector {
+
+    private final Object constant;
+
+    private ConstantLongColumnVongector(Object constant) {
+      this.constant = constant;
+    }
+
+    @Override
+    public long getLong(int i) {
+      return (long) constant;
+    }
+
+    @Override
+    public boolean isNullAt(int i) {
+      return constant == null;
+    }
+  }
+
+  private static class ConstantBooleanColumnVector implements BooleanColumnVector {
+    private final Object constant;
+
+    private ConstantBooleanColumnVector(Object constant) {
+      this.constant = constant;
+    }
+
+    @Override
+    public boolean getBoolean(int i) {
+      return (boolean) constant;
+    }
+
+    @Override
+    public boolean isNullAt(int i) {
+      return constant == null;
+    }
+  }
+
+  private static class ConstantDoubleColumnVector implements DoubleColumnVector {
+    private final Object constant;
+
+    private ConstantDoubleColumnVector(Object constant) {
+      this.constant = constant;
+    }
+
+    @Override
+    public boolean isNullAt(int i) {
+      return constant == null;
+    }
+
+    @Override
+    public double getDouble(int i) {
+      return (double) constant;
+    }
+  }
+
+  private static class ConstantFloatColumnVector implements FloatColumnVector {
+    private final Object constant;
+
+    private ConstantFloatColumnVector(Object constant) {
+      this.constant = constant;
+    }
+
+    @Override
+    public float getFloat(int i) {
+      return (float) constant;
+    }
+
+    @Override
+    public boolean isNullAt(int i) {
+      return constant == null;
+    }
+  }
+
+  private static class ConstantDecimalColumnVector implements DecimalColumnVector {
+    private final Object constant;
+
+    private ConstantDecimalColumnVector(Object constant) {
+      this.constant = constant;
+    }
+
+    @Override
+    public DecimalData getDecimal(int i, int precision, int scale) {
+      return (DecimalData) constant;
+    }
+
+    @Override
+    public boolean isNullAt(int i) {
+      return constant == null;
+    }
+  }
+
+  private static class ConstantTimestampColumnVector implements TimestampColumnVector {
+    private final Object constant;
+
+    private ConstantTimestampColumnVector(Object constant) {
+      this.constant = constant;
+    }
+
+    @Override
+    public TimestampData getTimestamp(int i, int precision) {
+      return (TimestampData) constant;
+    }
+
+    @Override
+    public boolean isNullAt(int i) {
+      return constant == null;
+    }
+  }
+
+  private static class ConstantBytesColumnVector implements BytesColumnVector {
+    private final Object constant;
+
+    private ConstantBytesColumnVector(Object constant) {
+      this.constant = constant;
+    }
+
+    @Override
+    public Bytes getBytes(int i) {
+      BinaryStringData str = (BinaryStringData) constant;
+      return new Bytes(str.toBytes(), 0, str.getSizeInBytes());
```

Comment:
The `str.toBytes()` will create a new object byte array each time, for a constant value  it's not worth to new byte[] every time.  I will suggest to allocate that data bytes in the constructor, and then always return the same instance.

Reply:
I read the code of `str.toBytes()` ,and I found that it is still more complicated, and it have done a lot of operations. I have not found a suitable method to assign different `BinaryStringData` to the same byte[].